src/app/pages/interest/interest.js

```diff
@@ -1,178 +1,189 @@
+                toggleBook={toggleBook}
+            />
+            <FormInput
+                longLabel="How many students do you teach each semester?"
+                inputProps={{
+                    name: 'Number_of_Students__c',
+                    type: 'number',
+                    min: '1',
+                    max: '999',
+                    required: true
+                }}
+            />
+            <HowDidYouHear />
+            {
+                bookBeingSubmitted &&
+                    <React.Fragment>
+                        <input type="hidden" name="Subject__c" value={bookBeingSubmitted.value} />
+                        <input type="hidden" name="First__c" value={bookBeingSubmitted === selectedBooks[0]} />
+                    </React.Fragment>
+            }
+        </div>
+    );
+}
+
+function FacultyForm({selectedRole, onPageChange}) {
+    const salesforce = useDataFromPromise(salesforcePromise, {});
+    const contactValidatorRef = useRef();
+    const selectedBooksRef = useRef();
+    const [currentBook, setCurrentBook] = useState();
+    const formRef = useRef();
+
+    function validatePage(page) {
+        const validateContactInfo = contactValidatorRef.current;
+
+        if (page === 1 && !validateContactInfo()) {
+            return false;
+        }
+        return true;
     }
 
-    firstPage() {
-        const contactForm = new ContactInfo({
-            validationMessage(name) {
-                return this.validated ? this.el.querySelector(`[name="${name}"]`).validationMessage : '';
+    useEffect(() => {
+        if (currentBook) {
+            formRef.current.submit();
+        }
+    }, [currentBook]);
+
+    function onSubmit(form) {
+        const selectedBooks = selectedBooksRef.current;
+        const submitQueue = selectedBooks.slice();
+        const iframe = document.getElementById(form.target);
+
+        // eslint-disable-next-line no-use-before-define
+        const submitAfterDelay = () => setTimeout(submitNextBook, 300);
+
+        function submitNextBook() {
+            const preselectedTitle = firstSearchArgument();
+
+            if (submitQueue.length > 0) {
+                setCurrentBook(submitQueue.shift());
+            } else if (iframe) {
+                setCurrentBook(null);
+                iframe.removeEventListener('load', submitAfterDelay);
+                afterFormSubmit(preselectedTitle, selectedBooks);
             }
-        });
-        const result = new SeriesOfComponents({
-            className: 'page-1',
-            contents: [
-                this.hiddenFields,
-                contactForm
-            ]
-        });
-
-        result.validate = function () {
-            return contactForm.checkSchoolName() || contactForm.validate();
-        };
-        return result;
+        }
+
+        formRef.current = form;
+        if (iframe) {
+            iframe.addEventListener('load', submitAfterDelay);
+        }
+        submitNextBook();
```

Comment:
There's the possibility of a bug here if the form is unmounted in the 300ms after `submitAfterDelay` is called. I think you can work around that by checking if one of the references are still valid in `submitNextBook`.  I think that references are reset to null when unmounting, but not entirely sure about that.  

To test you could increase the timeout to a large number like 10 seconds, submit and then hit browser back button, then watch the console to see what happens.

Reply:
It looks like `selectedBooks` got cleared when the form unmounted, so the queue became empty. No complaints in the Console, at least.